octopusdeploy/schema_version_control_settings.go

```diff
@@ -1,78 +1,46 @@
 package octopusdeploy
 
 import (
+	"net/url"
+
 	"github.com/OctopusDeploy/go-octopusdeploy/octopusdeploy"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-func expandVersionControlSettings(flattenedVersionControlSettings interface{}) *octopusdeploy.VersionControlSettings {
-	versionControlSettingsList := flattenedVersionControlSettings.(*schema.Set).List()
-	versionControlSettingsMap := versionControlSettingsList[0].(map[string]interface{})
-
-	versionControlSettings := &octopusdeploy.VersionControlSettings{
-		BasePath:      versionControlSettingsMap["base_path"].(string),
-		DefaultBranch: versionControlSettingsMap["default_branch"].(string),
-		HasValue:      versionControlSettingsMap["has_value"].(bool),
-		URL:           versionControlSettingsMap["url"].(string),
-		Username:      versionControlSettingsMap["username"].(string),
+func expandVersionControlSettings(values interface{}) *octopusdeploy.VersionControlSettings {
+	if values == nil {
+		return nil
 	}
 
-	if password := versionControlSettingsMap["password"].(string); len(password) > 0 {
-		versionControlSettings.Password = octopusdeploy.NewSensitiveValue(password)
+	flattenedValues := values.([]interface{})
+	if len(flattenedValues) == 0 || flattenedValues[0] == nil {
+		return nil
 	}
 
-	return versionControlSettings
-}
+	flattenedMap := flattenedValues[0].(map[string]interface{})
 
-func flattenVersionControlSettings(versionControlSettings *octopusdeploy.VersionControlSettings) []interface{} {
-	if versionControlSettings == nil {
-		return nil
+	if flattenedMap["type"] == "Database" {
```

Comment:
Could this be something like `octopusdeploy.NewDatabaseSettings()`, to be more consistent with the line in the return at the bottom of this method? Maybe that method doesn't need the "VersionControlled" parameter, if that's all it ever creates?